spec.md

```diff
@@ -170,8 +170,21 @@ that contains both context and data).
 * Constraints:
   * REQUIRED
   * MUST be a non-empty string
+  * MUST be a unique resource relative to source-service
 * Examples:
-  * my.s3.bucket
+  * mybucket.s3.amazonaws.com
+  * /projects/myProject/databases/(default)/documents/users/aNewUser
+
+### source-service
+* Type: String
+* Description: The physical (e.g. IP address) or logical (e.g. DNS address) of
+  the source software.
```

Comment:
In an environment where software components can move easily, how would someone use this field later on? In particular if its the IP.

Is the source-id not something that can be mapped to the source-service from within the source's environment?

I'm not asking to push back - I'm just trying to understand because it feels like this might be a dup of source-id to some extent.

Reply:
Were I using this in my own software I would strongly prefer something other than IP (e.g. DNS). I didn't want to require DNS as a prerequisite to an events system.

Reply:
Could the source service also be an instance id according to the [Open Service Broker API](https://www.openservicebrokerapi.org/)?

Reply:
yea, I would prefer a description that allowed more flexibility w.r.t. what the "source" considers useful to help identify where the event came from.  At least, that's what I view this (and other source-*) fields being used for. Is that consistent with others?

Reply:
Also, see this comment: https://github.com/cloudevents/spec/pull/68#issuecomment-365525662

Reply:
@deissnerk Absolutely. my intuition would be that the source-service would be the domain where the Service Broker were hosted and the source-id would be "/service_instances/:instance_id". Using a full resource path rather than just the ID would help us distinguish from an event, say, coming from the catalog when a new service was made available.

Reply:
@inlined Yes, that would be nice, but is "/service_instances/:instance_id" really a source-id here? For the source-id you have to refer to some object inside the service like a queue or a table. "/service_instances/:instance_id/:object_id" would be a confusing URI, as it would look like a URL.